nft-simple/src/nft_core.rs

```diff
@@ -20,35 +21,52 @@ pub trait NonFungibleTokenCore {
         &mut self,
         receiver_id: ValidAccountId,
         token_id: TokenId,
-        msg: String,
+        enforce_owner_id: Option<ValidAccountId>,
         memo: Option<String>,
+        msg: String,
     ) -> Promise;
 
+    fn nft_approve_account_id(&mut self, token_id: TokenId, account_id: ValidAccountId) -> bool;
+
+    fn nft_revoke_account_id(&mut self, token_id: TokenId, account_id: ValidAccountId) -> bool;
+
+    fn nft_revoke_all(&mut self, token_id: TokenId);
+
     fn nft_total_supply(&self) -> U64;
+
+    fn nft_token(&self, token_id: TokenId) -> Option<Token>;
 }
 
 #[ext_contract(ext_non_fungible_token_receiver)]
 trait NonFungibleTokenReceiver {
     /// Returns `true` if the token should be returned back to the sender.
     /// TODO: Maybe make it inverse. E.g. true to keep it.
```

Comment:
I was thinking about this one, too. I was wondering if we even need a return at all.
This is an interesting approach we can also consider from ERC-721:

>it calls `onERC721Received` on `_to` and throws if the return value is not `bytes4(keccak256("onERC721Received(address,address,uint256,bytes)"))`.

https://eips.ethereum.org/EIPS/eip-721

Reply:
Basically, we should `panic` if we don't want to receive. I can see a scenario where this is undesirable. Let's say the receiver contract needs to do an extra cross-contract operation before it can finalize the transfer. In this case, the extra operation requires a callback back on the receiver contract. But if you have to panic at the callback, then you can't revert the state of the receiver. This probably can be avoided, but a the same time, explicit `true` is more secure.

Reply:
Looks like this has been addressed